Hymns/Tags/TagSheetView.swift

```diff
@@ -0,0 +1,71 @@
+import SwiftUI
+import Resolver
+
+struct TagSheetView: View {
+
+    @ObservedObject private var viewModel: TagsViewModel
+    @Environment(\.presentationMode) var presentationMode
+    @State private var tagNames = ""
+    let favoritesStore: FavoritesStore
+    private var title: String
+    private var hymnIdentifier: HymnIdentifier
+
+    init(title: String, hymnIdentifier: HymnIdentifier, viewModel: TagsViewModel = Resolver.resolve(), favoritesStore: FavoritesStore = Resolver.resolve()) {
+        self.hymnIdentifier = hymnIdentifier
+        self.title = title
+        self.viewModel = viewModel
+        self.favoritesStore = favoritesStore
+    }
+
+    var body: some View {
+        Form {
+            Section {
+                HStack {
+                    Text("Label your tag")
+                    Spacer()
+                    Button(action: {
+                        self.presentationMode.wrappedValue.dismiss()
+                    }, label: {
+                        Image(systemName: "xmark.circle")
+                    })
+                }
+                HStack {
+                    TextField("Add tags", text: self.$tagNames)
+                    Button("Add") {
+                        self.favoritesStore.storeFavorite(FavoriteEntity(hymnIdentifier: self.hymnIdentifier, songTitle: self.title, tags: self.tagNames))
+                        self.viewModel.fetchTagsByHymn(hymnSelected: self.hymnIdentifier)
+                    }
+                }
+            }
+            Section {
+                HStack {
+                    ForEach(self.viewModel.tags, id: \.self) { tag in
+                        Button(action: {
+                            //TODO FIX Delete is deleting all the tags because the button when clicked is clicking all of them.
```

Comment:
Unfortunately, not. It's like a UI facing bug with the Buttons being in an HStack. Idk, maybe its even the simulator. But I will wait until we get our specs from Emily to address since that may change the way the UI here is structured